ts/store/reducers/entities/organizations/organizationsByFiscalCodeReducer.ts

```diff
@@ -23,6 +26,20 @@ const reducer = (
   action: Action
 ): OrganizationNamesByFiscalCodeState => {
   switch (action.type) {
+    // when this action is performed, all the keys (fiscal code of the organization)
+    // that are not present in the payload are removed from the state.
+    case getType(deleteUselessOrganizations):
+      return Object.keys(state)
+        .filter(key => action.payload.indexOf(key) !== -1)
+        .reduce<OrganizationNamesByFiscalCodeState>(
```

Comment:
Not sure about performance, but you could also check if `action.payload.indexOf(key) !== -1` as condition on the return of the reducer. Something like:

````
return Object.keys(state)
        .reduce<OrganizationNamesByFiscalCodeState>(
          (acc: OrganizationNamesByFiscalCodeState, key) => {
            if (action.payload.indexOf(key) !== -1){
            return {
              ...acc,
              [key]: state[key]
            }}
            else {
              return acc
            }
          },
          {}
        );

````